.github/workflow/ci.yaml

```diff
@@ -0,0 +1,39 @@
+name: Go
+
+on: [push, pull_request]
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
+      - name: Vet
+        run: go vet $(go list ./... | grep -v /cmd/wasm)
+
+      - name: Unit Tests
+        run: go test -v $(go list ./... | grep -v /cmd/wasm) -coverprofile=coverage.out
+
+      - name: Check Test coverage
+        env:
+          TEST_COVERAGE_THRESHOLD: 80
+        run: |
+          echo "Check if test coverage is above threshold ..."
+          echo "Threshold             : $TEST_COVERAGE_THRESHOLD %"
+          totalCoverage=`go tool cover -func=coverage.out | grep total | grep -Eo '[0-9]+\.[0-9]+'`
+          echo "Current test coverage : $totalCoverage %"
+          if (( $(echo "$totalCoverage $TEST_COVERAGE_THRESHOLD" | awk '{print ($1 > $2)}') )); then
+            echo "OK"
+          else
+            echo "Current test coverage is below threshold. Please add more unit tests."
+            echo "Failed"
+            exit 1
+          fi
+
+      - name: Build
+        run: go build -v $(go list ./... | grep -v /cmd/wasm)
```

Comment:
**question:** what is this build step there for? Is it simply meant as an assurance that the code compiles?

Reply:
I don't have strong feelings regarding this, I only leave it there since build is in the example of github ci workflow regarding go.